### Session : 상품구매한 방문자의 세션에 관한 정보

In [1]:
sess = pd.read_csv("../../data/05_Session.csv")
sess.tail()

CLNT_ID  SESS_ID  SESS_SEQ   SESS_DT  TOT_PAG_VIEW_CT TOT_SESS_HR_V  \
2712902  3500464  8713975       308  20180508             71.0           423   
2712903  3501155  4636571        86  20180714             44.0         2,090   
2712904  3501397   582632       128  20180920            164.0         2,435   
2712905  3573271  2328407         9  20180822              5.0           986   
2712906  3573758  8702345        17  20180508             18.0           373   

        DVC_CTG_NM             ZON_NM      CITY_NM  
2712902     mobile              Seoul        Seoul  
2712903    desktop              Seoul        Seoul  
2712904     mobile              Seoul        Seoul  
2712905    desktop              Busan        Busan  
2712906    desktop  Chungcheongbuk-do  Cheongju-si

In [3]:
for i in sess.columns[2:]:
    print("column : {}, unique count : {}".format(i, sess[i].nunique()))
    print(sess[i].unique())
    print()

column : SESS_SEQ, unique count : 3639
[   9   21   40 ... 3371 3050 2771]

column : SESS_DT, unique count : 183
[20180509 20180611 20180624 20180913 20180412 20180920 20180903 20180902
 20180904 20180413 20180930 20180925 20180923 20180825 20180716 20180606
 20180905 20180927 20180511 20180603 20180521 20180727 20180731 20180609
 20180510 20180807 20180822 20180910 20180912 20180908 20180907 20180915
 20180922 20180919 20180529 20180528 20180928 20180831 20180610 20180416
 20180514 20180824 20180918 20180917 20180805 20180617 20180613 20180827
 20180424 20180422 20180530 20180604 20180812 20180620 20180826 20180828
 20180420 20180901 20180411 20180926 20180816 20180503 20180502 20180430
 20180724 20180705 20180914 20180505 20180419 20180618 20180829 20180508
 20180629 20180820 20180526 20180730 20180519 20180616 20180818 20180408
 20180916 20180405 20180710 20180507 20180715 20180814 20180924 20180628
 20180729 20180823 20180627 20180723 20180720 20180929 20180830 20180909
 20180709 2

In [5]:
sess[sess.CLNT_ID==3573758]

CLNT_ID  SESS_ID  SESS_SEQ   SESS_DT  TOT_PAG_VIEW_CT TOT_SESS_HR_V  \
2607263  3573758  9206743         6  20180430             26.0         3,711   
2607264  3573758   776159        37  20180917             58.0         1,801   
2712906  3573758  8702345        17  20180508             18.0           373   

        DVC_CTG_NM             ZON_NM      CITY_NM  
2607263    desktop  Chungcheongbuk-do  Cheongju-si  
2607264    desktop  Chungcheongbuk-do  Cheongju-si  
2712906    desktop  Chungcheongbuk-do  Cheongju-si

### 궁금증
- 세션ID의 순서를 알 수 있다면, SESS_SEQ=37일때 1~37까지 나와야하는거 아닌가? 중간에 없는건 뭐지... --> 상품구매한 세션만 기록된 것임(해결)
- 연령대별로 어떤 기기에서 접속했는지 custom과 조인해보면 알 수 있지 않을까?, 지역도 마찬가지!
- SESS_DT를 가지고 구매주기를 계산해볼 수 있지 않을까?
- 위 테이블에서 약 1시간(3,711초)동안 총 26개의 페이지를 보고 구매에 이르렀다는 말인가? 